src/main/java/org/onebusaway/alexa/handlers/intent/YesIntentHandler.java

```diff
@@ -0,0 +1,91 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.onebusaway.alexa.handlers.intent;
+
+import com.amazon.ask.model.Response;
+import lombok.extern.log4j.Log4j;
+import org.apache.commons.lang3.StringUtils;
+import org.onebusaway.alexa.constant.Prompt;
+import org.onebusaway.alexa.constant.SessionAttribute;
+import org.onebusaway.alexa.exception.OneBusAwayException;
+import org.onebusaway.alexa.lib.ObaUserClient;
+import org.onebusaway.alexa.storage.ObaUserDataItem;
+import org.onebusaway.alexa.util.CityUtil;
+import org.onebusaway.alexa.util.RouteFilterUtil;
+import org.onebusaway.alexa.util.StopUtil;
+
+import java.util.Optional;
+
+import static org.onebusaway.alexa.constant.SessionAttribute.ASK_STATE;
+
+/**
+ * Handler for Yes intent request(e.g. user say "Yes" during the dialog).
+ */
+@Log4j
+public class YesIntentHandler extends IntentHandler {
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getIntentRequestName() {
+        return "AMAZON.YesIntent";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Optional<Response> handleWithObaData(final ObaUserDataItem obaUserDataItem, final ObaUserClient obaUserClient) {
+        if (askState == SessionAttribute.AskState.VERIFYSTOP) {
+            addOrUpdateSessionAttribute(ASK_STATE, askState);
+            return handleWithoutObaData();
+        }
+
+        if (askState == SessionAttribute.AskState.FILTER_INDIVIDUAL_ROUTE) {
+            return RouteFilterUtil.handleFilterRouteResponse(this.attributesManager, true, obaDao, obaUserDataItem);
+        }
+
+        log.error("Received yes intent without a question.");
+        return promptHelper.getResponse(Prompt.GENERAL_ERROR_MESSAGE);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Optional<Response> handleWithoutObaData() {
+        if (askState == SessionAttribute.AskState.VERIFYSTOP) {
+            return StopUtil.handleDuplicateStopResponse(personalization.getPrincipleId(), attributesManager, true, googleMaps, obaClient, obaDao);
+        }
+
+        if (askState == SessionAttribute.AskState.COPY_PROFILE_CONFIRM && personalization.isPersonalized()) {
+            try {
+                Optional<ObaUserDataItem> obaUserDataItem = this.obaDao.getUserData(personalization.getUserId());
+                obaUserDataItem.get().setUserId(personalization.getPersonId());
+                obaUserDataItem.get().setPreviousResponse(StringUtils.EMPTY);
```

Comment:
I'm not seeing the `PreviousResponse` field be updated in the database in any case, which is used for the `AMAZON.RepeatIntent`. This field should always store the last response that was sent to the user so it can be repeated.

#### Steps to repro

1. Open the OneBusAway skill
1. Look at the database

#### What I see

The `PreviousResponse` field doesn't reflect what Alexa just told me

#### What I expect

The `PreviousResponse` field should contain a copy of what Alexa just told me

Reply:
This was intentionally remove the `PreviousResponse` for copy profile case. (because of the name SSML was not there and person may not want the same response).

We are saving the previous response in following cases based on the previous version
* [GET_CITY](https://github.com/OneBusAway/onebusaway-alexa/blob/537501d76545c97a81f2734367b351d9e364a624/src/main/java/org/onebusaway/alexa/AuthedSpeechlet.java#L134)
* [GET_STOP_NUMBER](https://github.com/OneBusAway/onebusaway-alexa/blob/537501d76545c97a81f2734367b351d9e364a624/src/main/java/org/onebusaway/alexa/AuthedSpeechlet.java#L148)
* [ENABLE_CLOCK_TIME and DISABLE_CLOCK_TIME](https://github.com/OneBusAway/onebusaway-alexa/blob/537501d76545c97a81f2734367b351d9e364a624/src/main/java/org/onebusaway/alexa/util/StorageUtil.java#L240)
* [YES and NO for AskState.FILTER_INDIVIDUAL_ROUTE state](https://github.com/OneBusAway/onebusaway-alexa/blob/537501d76545c97a81f2734367b351d9e364a624/src/main/java/org/onebusaway/alexa/util/RouteFilterUtil.java#L125)

The [YES and NO for AskState.FILTER_INDIVIDUAL_ROUTE state](https://github.com/OneBusAway/onebusaway-alexa/blob/537501d76545c97a81f2734367b351d9e364a624/src/main/java/org/onebusaway/alexa/util/RouteFilterUtil.java#L125) is missing in the current code path, will add it back

Reply:
Removing previous response for copy profile case is fine.

Why wouldn't we want the previous response for the default case of getting arrival times? I could see a user wanting to request this in a few use cases, including when a bus just departed - the user may want to repeat that info, rather than always request fresh arrivals which might drop that bus off the list.

Reply:
Wait - are you saying that previously (i.e., in the current production skill release) we didn't support the `PreviousResponse` for getting arrival times?

Reply:
> Wait - are you saying that previously (i.e., in the current production skill release) we didn't support the `PreviousResponse` for getting arrival times?

I think we do support this in `GET_ARRIVALS`.

> Why wouldn't we want the previous response for the default case of getting arrival times? I could see a user wanting to request this in a few use cases, including when a bus just departed - the user may want to repeat that info, rather than always request fresh arrivals which might drop that bus off the list.

I think I also save the previous response, let me following the step to reproduce

Reply:
The response contains the text for `ANNOUNCED_FEATURES_V1_1_0`.
Seems this feature was not being used anymore and it has been set to 1 after user onboarded, I will remove it to align with the current experience